File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.py
class Solution(object):
    def maxScore(self, cardPoints, k):
        """
        :type cardPoints: List[int]
        :type k: int
        :rtype: int
        """
        n = len(cardPoints)

        total_pts = sum(cardPoints)

        wind_size = n-k

        curr_sum = sum(cardPoints[0:wind_size])
        min_sum = curr_sum

        for i in range(wind_size,n):
            curr_sum += cardPoints[i]
            curr_sum -= cardPoints[i-wind_size]

            min_sum = min(min_sum,curr_sum)

        return total_pts - min_sum

File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int lsum = 0;
        int rsum = 0;
        int max_sum = 0;

        for(int i=0;i<k;i++){
            lsum += cardPoints[i];
        }
        max_sum = lsum;
        int j = cardPoints.size()-1;

        for(int i=k-1;i>=0;i--){
            lsum -= cardPoints[i];
            rsum += cardPoints[j];
            j--;
            max_sum = max(max_sum,lsum+rsum);
        }
        return max_sum;
    }
};